service-commands/src/commands/users.js

```diff
@@ -157,15 +165,42 @@ User.getClockValuesFromReplicaSet = async libsWrapper => {
   return libsWrapper.getClockValuesFromReplicaSet()
 }
 
-/** Delay execution for n ms */
-function delay (ms) {
-  return new Promise(resolve => setTimeout(resolve, ms))
+const getLatestIndexedBlock = async (endpoint = DISCOVERY_NODE_ENDPOINT) => {
+  return (await axios({
+    method: 'get',
+    baseURL: endpoint,
+    url: '/health_check'
+  })).data.latest_indexed_block
 }
 
-/** Wrapper for custom delay time */
-async function waitForIndexing (waitTime = 5000) {
-  console.info(`Pausing ${waitTime}ms for discprov indexing...`)
-  await delay(waitTime)
+/**
+ * Wait for the discovery node to catch up to the latest block on chain up to a max
+ * indexing timeout of default 5000ms.
+ * @param {*} executeOne
+ * @param {*} libsWrapper
+ * @param {number} maxIndexingTimeout default 5000ms
+ */
+const waitForLatestBlock = async (libsWrapper) => {
```

Comment:
Why does this function live in two places?

Reply:
I was thinking about how since this functionality is necessary in two repos that it was okay for duplication. however, after further consideration that seems unnecessary. added this fn to service-commands and removed from mad-dog!